Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -507,6 +509,55 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_match)
     }
 }
 
+// 22.2.5.8 RegExp.prototype [ @@matchAll ] ( string ), https://tc39.es/ecma262/#sec-regexp-prototype-matchall
+JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_match_all)
+{
+    auto* regexp_object = this_object_from(vm, global_object);
+    if (!regexp_object)
+        return {};
+
+    auto string = vm.argument(0).to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    auto* constructor = species_constructor(global_object, *regexp_object, *global_object.regexp_constructor());
+    if (vm.exception())
+        return {};
+
+    auto flags_value = regexp_object->get(vm.names.flags);
+    if (vm.exception())
+        return {};
+    auto flags = flags_value.to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    bool global = flags.find('g').has_value();
+    bool unicode = flags.find('u').has_value();
```

Comment:
I actually did this one on purpose - I originally had it down below, but moved it up so I could do `move(flags)` just below here, rather than keeping a copy of `flags` alive.

Reply:
oh i see, my bad!